src/main/surfrdMod.F90

```diff
@@ -832,18 +826,24 @@ subroutine surfrd_veg_all(begg, endg, ncid, ns, actual_numcft)
                ' must also have a separate crop landunit, and vice versa)'//&
                errMsg(sourcefile, __LINE__))
     end if
```

Comment:
Rearranged a little here:
1) Moved wt_cft a few lines down without an if statement.
2) Moved both call check_sums a few lines down, one without an if and the other with if (.not. use_fates)
3) Kept call collapse_crop_types in the new if (.not. use_fates) 
which covers many more cases than the old if (use_crop)